### PR TITLE
[FIX] mass_mailing: Traceback on mass mailing list merging

### PR DESCRIPTION
**Pull request by abz-odoo**
What are the steps to reproduce your issue ?

    Try to merge two mailing lists

What is currently happening ?

    Traceback

opw-2504230

### CODE REVIEW

**Comment by abz-odoo** [comment]
@tde-banana-odoo

File: addons/mass_mailing/wizard/mailing_list_merge.py [hunk 1]
```diff
@@ -22,8 +22,8 @@ def default_get(self, fields):
         res = super(MassMailingListMerge, self).default_get(fields)
         src_list_ids = self.env.context.get('active_ids')
         res.update({
-            'src_list_ids': src_list_ids,
-            'dest_list_id': src_list_ids and src_list_ids[0] or False,
+            'src_list_ids': [Command.set(src_list_ids)],
```
Comment (trj-odoo) [inline addons/mass_mailing/wizard/mailing_list_merge.py:25]: Will not work in 14.0, only in master

**Comment by abz-odoo** [comment]
@nle-odoo

**Comment by tde-banana-odoo** [comment]
Strange it was not spotted before ... could you add a test case, failing without your fix and working with your fix ? You will probably have to use the Form tool to ease the fix writing.

Thanks,

**Review by nle-odoo — APPROVED** [review]
looks good

~~I would prefer to not have commands if possible (but as you wish, the orm should transform recordset in commands anyway), like:~~

```diff
     @api.model
     def default_get(self, fields):
         res = super(MassMailingListMerge, self).default_get(fields)
-        src_list_ids = self.env.context.get('active_ids')
+        src_list = self.env['mailing.list'].browse(self.env.context.get('active_ids'))
         res.update({
-            'src_list_ids': src_list_ids,
-            'dest_list_id': src_list_ids and src_list_ids[0] or False,
+            'src_list_ids': src_list,
+            'dest_list_id': src_list[:1],
         })
         return res
```
__edit:__ the ORM is expecting to receive commands

once you are happy with your test (that should fail before the fix):

robodoo delegate+

__edit:__

As a side note, as we discussed this doesn't happen before 14.0 because then we only did a default_get, in 14.0 we do an onchange (that itself do the default_get) and it's the onchange that is expecting to receive commands

**Review by nle-odoo — APPROVED** [review]
robodoo r+

**Comment by tde-banana-odoo** [comment]
Sorry for being that guy @robodoo r-, but please use meaningful variable names ...

File: addons/mass_mailing/tests/test_mailing_list.py [hunk 2]
```diff
@@ -72,13 +72,14 @@ def test_mailing_list_merge(self):
...
-        result_list = self.env['mailing.list.merge'].create({
-            'src_list_ids': [(4, list_id) for list_id in [self.mailing_list_1.id, self.mailing_list_2.id]],
-            'dest_list_id': self.mailing_list_3.id,
-            'merge_options': 'existing',
-            'new_list_name': False,
-            'archive_src_lists': False,
-        }).action_mailing_lists_merge()
+        list_merge = self.env['mailing.list.merge'].with_context(active_ids=[self.mailing_list_1.id, self.mailing_list_2.id])
+
+        with Form(list_merge) as f:
```
Comment (tde-banana-odoo) [inline addons/mass_mailing/tests/test_mailing_list.py:77]: Don't use "f"

File: addons/mass_mailing/tests/test_mailing_list.py [hunk 3]
```diff
@@ -72,13 +72,14 @@ def test_mailing_list_merge(self):
...
         # duplicates are appearing in C
 
-        result_list = self.env['mailing.list.merge'].create({
-            'src_list_ids': [(4, list_id) for list_id in [self.mailing_list_1.id, self.mailing_list_2.id]],
-            'dest_list_id': self.mailing_list_3.id,
-            'merge_options': 'existing',
-            'new_list_name': False,
-            'archive_src_lists': False,
-        }).action_mailing_lists_merge()
+        list_merge = self.env['mailing.list.merge'].with_context(active_ids=[self.mailing_list_1.id, self.mailing_list_2.id])
```
Comment (tde-banana-odoo) [inline addons/mass_mailing/tests/test_mailing_list.py:75]: Could be inlined in Form call

File: addons/mass_mailing/tests/test_mailing_list.py (hunk 2 above)
Comment (tde-banana-odoo) [inline addons/mass_mailing/tests/test_mailing_list.py:77]: Actually why using a with statement at all ? Makes code complicated to read ... you could simply have
```
merge_form = Form() ...
```

**Review by tde-banana-odoo — COMMENTED** [review]
Coding style !

**Comment by nle-odoo** [comment]
thanks

`with` only save automatically from what I can see, here we saved manually anyway

robodoo r+

**Comment by tde-banana-odoo** [comment]
Thanks for updating !

**Comment by sswapnesh** [comment]
I already made PR for the fix a long ago https://github.com/odoo/odoo/pull/62980

**Pull request closed without merging**